docs(account-settings): delete user docs (#3087)
aws-amplify/amplify-ui: docs/src/pages/[platform]/connected-components/account-settings/delete-user/react.mdx

<ReactPropsTable props={OVERRIDES} />

#### Reusing default components

Default components are accessible through `DeleteUser.COMPONENT_NAME` (e.g. `DeleteUser.NewPassword`) for your convenience. This is useful if you're interested in modifying just a small part of UI instead of overriding the whole component.

```jsx
function App() {
  const components = {
    // reuse default component
    SubmitButton: (props) => (
      <DeleteUser.SubmitButton>Custom Submit Button</DeleteUser.SubmitButton>
    ),
  }

  return (
    <DeleteUser components={components} />
  );
}
```

#### Using your own component

You can also provide your own custom to override components. 

```jsx
function App() {
  const components = {
    // use your own component
    SubmitButton: (props) => (
      <button {...props} type="submit">
        Custom Button Text
      </button>
    )
  };
  
  return (
    <DeleteUser components={components} />
  );
}
```

Note that spreading props is necessary so that event listeners like `onChange`` or html attributes like `name`` are attached correctly. If you're providing your own custom components, make sure props are spread onto your component.

> #### Overriding warning component

Author: ```suggestion
#### Overriding warning component

```

### Props

<ReactPropsTable props={DELETE_USER} />

### Basic Usage

`DeleteUser` has `onSuccess` handler that will be called after successful user account deletion. You may use this callback to run any custom handling, such as navigating route or rendering confirmation messages.

```jsx
import React from 'react';
import { DeleteUser } from '@aws-amplify/ui-react';

function App() {
  const handleSuccess = () => {
    alert('password is successfully changed!')
  }
  
  return (
    <DeleteUser onSuccess={handleSuccess}/>
  );
}
```

### Override default delete handler
You can override the default deleteUser handler with `handleDelete` prop to add any custom cleanup handlers before or after account deletion.

```jsx
import React from 'react';

import { Auth } from 'aws-amplify';
import { DeleteUser } from '@aws-amplify/ui-react';

function App() {
  const handleSuccess = () => {
    alert('password is successfully changed!')
  }

  const handleDelete = async () => {
    console.log("Doing some clean up...");
    console.log("Done!");
    await Auth.deleteUser();
  };  
      
  return (
    <DeleteUser onSuccess={handleSuccess}/>

>   );

Reviewer: Does `handleDelete` not need to be passed into the `DeleteUser` component?

Author: Thanks for the catch 🤦

```suggestion
You can override the default deleteUser handler with `handleDelete` prop to add any custom cleanup handlers before or after account deletion.

```jsx
import React from 'react';

import { Auth } from 'aws-amplify';
import { DeleteUser } from '@aws-amplify/ui-react';

function App() {
  const handleSuccess = () => {
    alert('password is successfully changed!')
  }

  const handleDelete = async () => {
    console.log("Doing some clean up...");
    console.log("Done!");
    await Auth.deleteUser();
  };  
      
  return (
    <DeleteUser onSuccess={handleSuccess}/ handleDelete={handleDelete}>
  );
```

>     <DeleteUser onSuccess={handleSuccess}/ handleDelete={handleDelete}>

Author: ```suggestion
    <DeleteUser onSuccess={handleSuccess} handleDelete={handleDelete} />
```

}
```

### Component Overrides

You can provide your own UI components to override parts of `DeleteUser`. It supports the following slots:

<ReactPropsTable props={OVERRIDES} />

#### Reusing default components

Default components are accessible through `DeleteUser.COMPONENT_NAME` (e.g. `DeleteUser.NewPassword`) for your convenience. This is useful if you're interested in modifying just a small part of UI instead of overriding the whole component.

```jsx
function App() {
  const components = {
    // reuse default component
    SubmitButton: (props) => (

>       <DeleteUser.SubmitButton>Custom Submit Button</DeleteUser.SubmitButton>

Author: ```suggestion
      <DeleteUser.SubmitButton {...props}>Custom Submit Button</DeleteUser.SubmitButton>
```

    ),
  }

  return (
    <DeleteUser components={components} />
  );
}
```

#### Using your own component

> You can also provide your own custom to override components. 

Reviewer: ```suggestion
You can also provide your own custom buttons to override components. 
```

Reviewer: or `elements` ?

Author: Oh, good catch, thanks! I'll go with

```suggestion
You can override parts of the `DeleteUser` component by passing in a `components` prop.
```

that I stole from initial FileUploader doc 😈

```jsx
function App() {
  const components = {
    // use your own component
    SubmitButton: (props) => (
      <button {...props} type="submit">
        Custom Button Text
      </button>
    )
  };
  
  return (
    <DeleteUser components={components} />
  );
}
```

> Note that spreading props is necessary so that event listeners like `onChange` or html attributes like `name` are attached correctly. If you're providing your own custom components, make sure props are spread onto your component.

Reviewer: ```suggestion
Note that spreading props is necessary so that event listeners like `onChange` or html attributes like `name` are attached correctly. If you're providing your own custom components, make sure props are spread onto it.
```

Reviewer: I just don't like repeating the word `components` twice in this sentence.

Author: I like the change!

> Note that spreading props is necessary so that event listeners like `onChange`` or html attributes like `name`` are attached correctly. If you're providing your own custom components, make sure props are spread onto your component.

Author: ```suggestion
Note that spreading props is necessary so that event listeners like `onChange` or html attributes like `name` are attached correctly. If you're providing your own custom components, make sure props are spread onto your component.
```

#### Overriding warning component

Warning component requires the following props:

<ReactPropsTable props={WARNING} />

The following example overrides the warning component to require users to type "delete" before they can delete their account.

```jsx
import React from 'react';
import { 
  Button,
  DeleteUser,  
  Flex,
  Text,
  TextField,
} from '@aws-amplify/ui-react';

function CustomWarning({onCancel, onConfirm, isDisabled}) {
  const [value, setValue] = React.useState('');
  
  const handleChange = (event) => {
    event.preventDefault();

    const { value } = event.target;
    setValue(value);
  }

  return (

>     <form onSubmit={onSubmit}>

Reviewer: ```suggestion
          <View as="form" onSubmit={onSubmit}>
```

Author: This does not work due to a known form: https://github.com/aws-amplify/amplify-ui/issues/2891

Reviewer: Good to know!

import { Alert, Expander, ExpanderItem, ScrollView } from '@aws-amplify/ui-react';

import ReactPropsTable from '@/components/propsTable/ReactPropsTable';
import { Example, ExampleCode } from '@/components/Example';
import { DELETE_USER, OVERRIDES, WARNING, DELETE_BUTTON } from './props';
import { ERROR_MESSAGE } from '../props';

### Props

<ReactPropsTable props={DELETE_USER} />

### Basic Usage

`DeleteUser` has `onSuccess` handler that will be called after successful user account deletion. You may use this callback to run any custom handling, such as navigating route or rendering confirmation messages.

<Example>
  <ExampleCode>

```jsx
import React from 'react';
import { AccountSettings } from '@aws-amplify/ui-react';

function App() {
  const handleSuccess = () => {
    alert('password is successfully changed!')
  }
  
  return (
    <AccountSettings.DeleteUser onSuccess={handleSuccess} />
  );
}
```

  </ExampleCode>
</Example>

### Override default delete handler

> You can override the default deleteUser handler with `handleDelete` prop to add any custom cleanup handlers before or after account deletion.

Reviewer 2: nit: "`deleteUser` handler"

Author: ```suggestion
You can override the default `deleteUser` handler with `handleDelete` prop to add any custom cleanup handlers before or after account deletion.
```

Author: `radii` for `medium` changed in 65388d0a1

    <ReactPropsTable props={ERROR_MESSAGE} />

The following example overrides the error message component with [`Heading`](/components/heading) and [`Text`](/components/text) primitives.

    <Example>
      <ExampleCode>

```jsx
import { Heading, Text } from '@aws-amplify/ui-react';

function App() {
  const components = {
    ErrorMessage: ({ children }) => (
      <>
        <Heading>Custom Error Message</Heading>
        <Text variation="warning">{children}</Text>
      </>
    )
  };
  
  return (
    <AccountSettings.DeleteUser components={components} />
  );
}
```

      </ExampleCode>
    </Example>
  </ExpanderItem>
  <ExpanderItem title="WarningView" value="warning-view">

Props:

    <ReactPropsTable props={WARNING} />

The following example overrides the warning component to require users to type "delete" before they can delete their account.

    <ScrollView height="600px">
      <Example>
        <ExampleCode>
  
```jsx{10-37,48} file=../../../../../../../examples/next/pages/ui/components/account-settings/delete-user-with-overrides/index.page.tsx#L1-L72

```

        </ExampleCode>
      </Example>
    </ScrollView>
  </ExpanderItem>
</Expander>

### Theming

> `DeleteUser` component is built upon our robust [Amplify UI theming system](../../theming). To get started add a theme object and set the appropriate [design tokens](../../theming#design-tokens). You can then pass that `theme` to the `ThemeProvider` so the changes can take affect.

Reviewer 2: Same comments from ChangePassword PR ([one](https://github.com/aws-amplify/amplify-ui/pull/3085#discussion_r1040102043), [two](https://github.com/aws-amplify/amplify-ui/pull/3085#discussion_r1040102803), [three](https://github.com/aws-amplify/amplify-ui/pull/3085#discussion_r1040103050))

Author: ```suggestion
`DeleteUser` component is built upon our robust [Amplify UI theming system](../../theming). To get started, add a theme object and set the appropriate [design tokens](../../theming#design-tokens). You can then pass that `theme` to the `ThemeProvider` so the changes can take affect.
```
